update lang/branch (#75)
OI-wiki/i18n: docs/lang/branch.md

> By default, a program is executed following the order of code. Sometimes when certain statements need to be selectively executed, we can use the the branch function to achieve such flow control. Choosing the appropriate branch statement can improve the efficiency of the program.

Reviewer: branch (function)->(feature)

## if statement

### Basic if statement

> The following is the structure of a basic if statement.

Reviewer: ```suggestion
The following code shows the structure of a basic if statement.
```

```cpp
if (condition) {
  main body;
}
```

> The if statement calculates the condition to check: if the result is true (not 0), the statement is executed, otherwise it is not executed.

Reviewer: nonzero

If there is only a single statement in the body, the curly braces can be omitted.

### if...else statement

```cpp
if (condition) {
  statement1;
} else {
  statement2;
}
```

> The if...else statement is similar to the if statement, and the else does not need to write more conditions. When the condition of the if statement is satisfied, the statement in the if will be executed. And when the condition of the if statement is not satisfied, the statement in the else will be executed. Similarly, when the main part has only one statement, the curly braces can be omitted.

Reviewer: 建议代码里的这些 if, else, break, switch 啥的用 mono （``） 包起来

Reviewer: ```suggestion
The if...else statement is similar to the if statement, and the else does not need to write more conditions. When the condition of the if statement is satisfied, the statement in the if will be executed. Otherwise, the statement in the else will be executed. Similarly, the curly braces can be omitted when the main part has only one statement.
```

In fact, this statement is equivalent to the else clause of the first if with only one if statement, so the curly braces are omitted and placed together. If the conditions are parallel to each other, writing this way can make the code logic clearer.

Logically, it is roughly equivalent to this passage:

> When solving a quadratic equation in one variable, the relationship between the roots of the equation and the discriminant:
>
> -   If ( $\Delta<0$ )
>     The equation has no solution;
> -   Otherwise, if ( $\Delta=0$ )
>     The equation has two identical real number solutions;
> -   Otherwise
>     The equation has two different real number solutions;

## switch statement

```cpp
switch (condition) {
  case label1:
    statement1;
  case label2:
    statement2;
  default:
    statement3;
}
```

> When the switch statement is executed, the value is selected first, and then the corresponding label is selected according to the value of the choice sentence. And the execution starts from the label. Among them, the choice sentence must be an integer type expression, and the label must be an integer type constant. E.g:

Reviewer: ```suggestion
When the switch statement is executed, the value is selected first, and then the corresponding label is selected according to the value of the choice sentence. And the execution starts from that label. Among them, the choice expression must be an integer type expression, and the label must be an integer type constant. E.g:
```

```cpp
char i = 'A';
// the data type of i here is character, but char

> // is also a type of integer and satisfies the requirements of an expression of integer type

Reviewer: // the data type of i here is character, but it is a kind of integer in
// nature and satisfies the requirement of an integer-typed expression

Reviewer: :thinking: Waiting for upstream discussion

}
```

> If you don't want the following branch statement to be executed, the break is required. For specific examples, please refer to the following code.

Reviewer: > 以上代码运行后输出的结果为 `WIKI` 和 `Hello World`
漏了这句
```suggestion
If you don't want the following branch statements to be executed, the break is required. For specific examples, please refer to the following code.
```

}
```

> The output of above code is WIKI, because the following statements will not continue to be executed since the break statement exists. The default statement does not need a break, because there is no statement below.

Reviewer: 输出加 mono 吧
```suggestion
The output of above code is WIKI, because the following statements will not continue to be executed for the 
 existence of break statement. The default statement does not need a break, because there is no statement below.
```

> It is optional to add curly braces in the case clause of switch. But it should be noted that if you need to define a variable in the switch statement, the curly braces must be added. E.g:

Reviewer: ```suggestion
It is optional to add curly braces in the case clause of switch. But note that the curly braces must be added if you need to define a variable in the switch statement. E.g:
```

}
```

??? note "How to understand switch"

>     In the above article, a lot of terms such as "case clause" and "case sub-clause" are used. In fact, in the underlying implementation, switch is equivalent to a set of jump statements. For this reason, clever technique like Duff's Device exists. Those who are interested to know more are welcome to learn by themselves.

Reviewer: ```suggestion
    In the article above, a lot of terms such as "case clause" and "case sub-clause" are used. In fact, in the underlying implementation, switch is equivalent to a set of jump statements. For this reason, clever techniques like Duff's Device exists. You can dig deeper about that if you are interested.
```

>     In the above article, a lot of terms such as "case clause" and "case sub-clause" are used. In fact, in the underlying implementation, switch is equivalent to a set of jump statements. For this reason, clever technique like Duff's Device exists. Those who are interested to know more are welcome to learn by themselves.

Reviewer: ```suggestion
    In the article above, a lot of terms such as "case clause" and "case sub-clause" are used. In fact, in the underlying implementation, switch is equivalent to a set of jump statements. For this reason, clever techniques like Duff's Device exists. You can dig deeper about that if you are interested.
```

??? note "How to understand switch"
    In the above article, a lot of terms such as "case clause" and "case sub-clause" are used. In fact, in the underlying implementation, switch is equivalent to a set of jump statements. For this reason, clever technique like Duff's Device exists. Those who are interested to know more are welcome to learn by themselves.

>     另外还有一种理解 switch 的方式，就是从其逻辑意义去理解。在这种理解里，case 代表的就是一组子句，而 switch 根据选择句的值选择某个 case 分句进行执行。在这种理解中，除了极少数例外，case 是必须要加 break 和花括号的。

Reviewer: Upstream update
